client/spa/js/instructor/spec/instructor.model.spec.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+/*
+global jasmine, describe, it, expect, beforeEach, afterEach, xdescribe, xit,
+spyOn
+*/
+// Get the code you want to test
+var Model = require('../instructor.model');
+
+// Test suite
+console.log('test instructor.model');
+describe('Instructor model ', function(){
+  var model;
+
+  describe('when creating a new model ', function(){
+    beforeEach(function(){
+      model = new Model();
+    });
+
+    xit('initializes with custom logic', function(){
+    });
+  });
+
+  describe('when updating the model for instructor ', function(){
+    var errorSpy;
+
+    beforeEach(function(){
```

Comment:
Try this:

beforeEach(function(){
      errorSpy = jasmine.createSpy('Invalid');
      model = new Model({
        id: 1,
        firstName: 'foo',
        lastName: 'foo',
        skills: 'foo'
      });
      model.on('invalid', errorSpy);
    });

    it('does not save when firstName is empty ', function(){
      model.set('firstName', null);
      model.save();
      expect(errorSpy).toHaveBeenCalled();
      expect(errorSpy.calls.mostRecent().args[0]).toBe(model);
      expect(errorSpy.calls.mostRecent().args[1][0]).toEqual(
        'firstName cannot be empty');
    });

    it('does not save when lastName is empty ', function(){
      model.set('lastName', null);
      model.save();
      expect(errorSpy).toHaveBeenCalled();
      expect(errorSpy.calls.mostRecent().args[0]).toBe(model);
      expect(errorSpy.calls.mostRecent().args[1][0]).toEqual(
        'lastName cannot be empty');
    });

    it('does not save when skills is empty ', function(){
      model.set('skills', null);
      model.save();
      expect(errorSpy).toHaveBeenCalled();
      expect(errorSpy.calls.mostRecent().args[0]).toBe(model);
      expect(errorSpy.calls.mostRecent().args[1][0]).toEqual(
        'skills cannot be empty');
    });